### PR TITLE
feat: handle CloudEvent and Message responses from function invocation

### PR DESCRIPTION
**Pull request by lance**
We already **kind of** handled them, but these changes make it more intuitive
for the user. For example, the function author can respond with `new CloudEvent(...)`
or `HTTP.binary(event)` and the framework should handle the response correctly.

This partially fixes https://github.com/boson-project/functions/issues/15

Signed-off-by: Lance Ball <lball@redhat.com>

### CODE REVIEW

File: lib/invoker.js [hunk 1]
```diff
@@ -7,10 +8,10 @@ module.exports = function invoker(func) {
       code: 200,
       response: undefined,
       headers: {
-        'Content-Type': 'application/json; charset=utf8',
-        'Access-Control-Allow-Methods':
+        'content-type': 'application/json; charset=utf8',
```
Comment (matejvasek) [inline lib/invoker.js:11]: Why this change, shouldn't it be case insensitive?

Comment (lance) [reply lib/invoker.js:11]: You're right - it should be, and it is here where we merge user headers with event headers https://github.com/boson-project/faas-js-runtime/pull/68/files#diff-89ae878b1a78765add577244271796f156213cdd0ab6722f641ef96926f38674R54.

That change to lowercase is kind of immaterial since we do lowercase all headers later. It was left over from some debugging I was doing when I realized that I was not doing case-insensitive comparisons. But I thought it would be fine to keep. :shrug:

File: lib/invoker.js [hunk 2]
```diff
@@ -47,9 +48,32 @@ module.exports = function invoker(func) {
...
+
+    // If the response is a CloudEvent, we need to convert it
+    // to a Message first and respond with the headers/body
+    if (payload.response instanceof CloudEvent) {
+      try {
+        const message = HTTP.binary(payload.response);
+        payload.headers = {...payload.headers, ...message.headers};
+        payload.response = message.body;  
+      } catch (e) {
+        console.error(e);
```
Comment (matejvasek) [inline lib/invoker.js:68]: Shouldn't we set `500` status code here, or is it handled somewhere else?

File: test/test.js [hunk 3]
```diff
@@ -226,25 +282,6 @@ test('Successfully handles events with no data', t => {
   });
 });
 
-test('Responds with error code (4xx or 5xx) to malformed cloud events', t => {
```
Comment (matejvasek) [inline test/test.js:229]: Why deleting this?

**Review by matejvasek — APPROVED** [review]

File: lib/invoker.js (hunk 2 above)
Comment (lance) [reply lib/invoker.js]: Ahh good catch! Fixed.

File: test/test.js (hunk 3 above)
Comment (lance) [reply test/test.js:229]: @matejvasek I will elaborate a little more here. The rationale behind this change is rooted in changes that were made in the upstream `cloudevents` module and which recently landed in v3.2.0. That version bump of the module introduced the `[HTTP.binary()](https://github.com/boson-project/faas-js-runtime/pull/68/files#diff-89ae878b1a78765add577244271796f156213cdd0ab6722f641ef96926f38674R54)` API method (and others) that I have made use of in this PR.

But a bigger thing that it did was change how the receiver handles malformed events. A summary of the wider community discussion on why receiving a CloudEvent should provide loose validation is summarized in [this comment](https://github.com/cloudevents/sdk-javascript/pull/328#issuecomment-682059664). 

Because of this change in the upstream validation behavior, an exception is never thrown on our end when receiving an event. In `cloudevents` 3.1.x it would have been and was, and was why this test worked. However, in order to maintain the existing behavior in this module, I would need to add a call to `event.validate()` when receiving them. To me that seems to land more in the user/customer domain. I thought a lot about this while making the upstream changes, and it's been pretty well hashed out in the community. So I think this module should follow the behavior upstream.
